cpp/src/sampling/detail/shuffle_and_organize_output.cu

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2022-2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "detail/graph_partition_utils.cuh"
+
+#include <cugraph/arithmetic_variant_types.hpp>
+#include <cugraph/detail/shuffle_wrappers.hpp>
+#include <cugraph/utilities/device_functors.cuh>
+
+#include <raft/core/handle.hpp>
+
+#include <rmm/device_uvector.hpp>
+
+#include <thrust/count.h>
+#include <thrust/gather.h>
+#include <thrust/sort.h>
+
+#include <optional>
+
+namespace cugraph {
+namespace detail {
+
+std::tuple<std::vector<cugraph::arithmetic_device_uvector_t>,
+           std::optional<rmm::device_uvector<int32_t>>,
+           std::optional<rmm::device_uvector<size_t>>>
+shuffle_and_organize_output(
+  raft::handle_t const& handle,
+  std::vector<cugraph::arithmetic_device_uvector_t>&& edges_with_properties,
```

Comment:
Just my 2 cents, `property_edges` to be little more succinct?

Reply:
I'll make that change.